leshan-server-core/src/main/java/org/eclipse/leshan/server/registration/RegistrationHandler.java

```diff
@@ -62,10 +63,16 @@ public RegisterResponse register(Identity sender, RegisterRequest registerReques
             return RegisterResponse.forbidden(null);
         }
 
-        Client client = new Client(RegistrationHandler.createRegistrationId(), registerRequest.getEndpointName(),
-                sender.getPeerAddress().getAddress(), sender.getPeerAddress().getPort(),
-                registerRequest.getLwVersion(), registerRequest.getLifetime(), registerRequest.getSmsNumber(),
-                registerRequest.getBindingMode(), registerRequest.getObjectLinks(), serverEndpoint);
+        Client.Builder builder = new Client.Builder(RegistrationHandler.createRegistrationId(),
+                registerRequest.getEndpointName(), sender.getPeerAddress().getAddress(), sender.getPeerAddress()
+                        .getPort(), serverEndpoint);
+
+        builder = builder.lwm2mVersion(registerRequest.getLwVersion()).lifeTimeInSec(registerRequest.getLifetime())
```

Comment:
builder = ...  is not needed
(our formatter does not really like inline syntax :p)